### DISCUSSION TITLE
Glob import and TypeScript

### DISCUSSION THREAD

**Discussion by samuliasmala** [parcel-bundler/parcel#6038] [Q&A] [answered]
I'm using Parcel 1.12.5 [glob import](https://parceljs.org/module_resolution.html#glob-file-paths) to load translation files. It works nicely but I get following TypeScript error in VS Code. It doesn't prevent the compilation but any advices how I could get rid of it?
```ts
// Cannot find module './localizations/*.json' or its corresponding type declarations.ts(2307)
import LocalizedResources from './localizations/*.json';
```

**Comment by FDiskas** [comment]
What about
```
declare module '*.json';
```

**Reply by FDiskas** [reply]
But actually 
```
 "compilerOptions": {
        "resolveJsonModule" : true,
    }
``` should work also

**Reply by samuliasmala** [reply]
I already have the `"resolveJsonModule" : true` in my config. Also the `declare module '*.json';` didn't help. I think both of those are for fixing issues importin a single json file. If I only import a single json I get no errors:

```js
// No TypeScript errors
import LocalizedResources from './localizations/en.json';
```

**Reply by FDiskas** [reply]
More info could be found in this issue https://github.com/jescalan/babel-plugin-import-glob-array/issues/7

**Comment by samuliasmala** [comment, marked as answer]
Thanks! I was now able to resolve the issue. My mistake was that the `declare module '*.json';` wasn't in `*.d.ts`file but in a regular `*.ts` file. after changing the filename extension the error was gone.

I ended up creating `declarations.d.ts` file with the following content:

```ts
declare module '*.json' {
  type fileContent = Record<string, unknown>;
  const files: Record<string, fileContent>;
  export default files;
}
```
